### PR TITLE
Adds Q/DQ layout support for embedding quantization with IntxWeightOnlyConfig

### PR DESCRIPTION
**Pull request by metascroy**
This will be used to quantize embeddings in ET.

### CODE REVIEW

File: torchao/experimental/quant_api.py [hunk 1]
```diff
@@ -638,6 +639,79 @@ def quantize(self, model: nn.Module) -> nn.Module:
...
+        group_size = granularity.group_size
+    elif isinstance(granularity, PerRow):
+        group_size = weight.shape[-1]
+    else:
+        raise ValueError(f"granularity must be PerGroup or PerRow, got {granularity}")
+
+    quant_min = -(1 << (bit_width - 1))
+    quant_max = (1 << (bit_width - 1)) - 1
+
+    weight = to_affine_quantized_intx_experimental(
```
Comment (metascroy) [inline torchao/experimental/quant_api.py:692]: TODO: change to_affine_quantized_intx_experimental to to_affine_quantized_intx.

File: torchao/experimental/quant_api.py [hunk 2]
```diff
@@ -638,6 +639,79 @@ def quantize(self, model: nn.Module) -> nn.Module:
...
+    scale_dtype = torch.float32
+
+    if isinstance(granularity, PerGroup):
+        group_size = granularity.group_size
+    elif isinstance(granularity, PerRow):
+        group_size = weight.shape[-1]
+    else:
+        raise ValueError(f"granularity must be PerGroup or PerRow, got {granularity}")
+
+    quant_min = -(1 << (bit_width - 1))
```
Comment (metascroy) [inline torchao/experimental/quant_api.py:689]: TODO: use defined quant dicts in quant_primitives

File: torchao/quantization/quant_api.py [hunk 3]
```diff
@@ -1569,6 +1572,92 @@ def _uintx_weight_only_transform(
     return module
 
 
+@dataclass
+class IntxWeightOnlyConfig(AOBaseConfig):
+    """
```
Comment (metascroy) [inline torchao/quantization/quant_api.py:1577]: @andrewor14 can you have a look at this comment if there are any issues with it working well with QAT workflow with FakeQuantizeConfig.

File: torchao/experimental/tests/test_embedding_xbit_quantizer.py [hunk 4]
```diff
@@ -134,14 +133,14 @@ def test_shared_embedding(self):
...
-            granularity=granularity,
+            granularity=PerAxis(0),
             has_weight_zeros=has_weight_zeros,
         ).quantize(quantized_model_reference)
         quantize_(
             quantized_model_reference,
             Int8DynamicActivationIntxWeightConfig(
                 weight_dtype=weight_dtype,
-                granularity=granularity,
+                granularity=PerRow(),
```
Comment (jerryzh168) [inline torchao/experimental/tests/test_embedding_xbit_quantizer.py:143]: should this be PerAxis as well

File: torchao/experimental/tests/test_embedding_xbit_quantizer.py [hunk 5]
```diff
@@ -155,7 +154,7 @@ def test_shared_embedding(self):
         quantized_model = copy.deepcopy(model)
         SharedEmbeddingQuantizer(
             weight_dtype=weight_dtype,
-            granularity=granularity,
+            granularity=PerRow(),
```
Comment (jerryzh168) [inline torchao/experimental/tests/test_embedding_xbit_quantizer.py:157]: and this

**Review by jerryzh168 — APPROVED** [review]
looks good overall, just need to change PerRow to PerAxis(axis=0) as we discussed in meeting

File: torchao/experimental/tests/test_embedding_xbit_quantizer.py (hunk 4 above)
Comment (metascroy) [reply torchao/experimental/tests/test_embedding_xbit_quantizer.py:143]: It can't be because that's controlled by Int8DynamicActivationIntxWeightConfig, which uses PerRow until https://github.com/pytorch/ao/pull/1968 lands

File: torchao/dtypes/affine_quantized_tensor_ops.py [hunk 6]
```diff
@@ -263,6 +269,9 @@ def _(func, types, args, kwargs):
 
 @implements(torch.nn.functional.embedding)
 def _(func, types, args, kwargs):
+    if _embedding_q_dq_check(args, kwargs):
+        return _embedding_q_dq_impl(args, kwargs)
+
```
Comment (kimishpatel) [inline torchao/dtypes/affine_quantized_tensor_ops.py:274]: why does line 299 only dequantizes weight bu tnot actually run embedding op?

File: torchao/quantization/quant_api.py (hunk 3 above)
Comment (kimishpatel) [reply torchao/quantization/quant_api.py:1579]: Strange that we have IntxWeightOnly and Int4WeightOnly

Comment (jerryzh168) [reply torchao/quantization/quant_api.py:1579]: yeah I feel we should probably merge these two
